vendor/github.com/docker/go-connections/tlsconfig/config.go

```diff
@@ -28,7 +31,8 @@ type Options struct {
 	// client-only option
 	InsecureSkipVerify bool
 	// server-only option
-	ClientAuth tls.ClientAuthType
+	ClientAuth          tls.ClientAuthType
+	PassphraseRetriever notary.PassRetriever `json:"-"`
```

Comment:
Rather than `go-connections` taking a passphrase retriever, it could take an optional passphrase for the key.  It'd be up to the caller to get it however they deem fit.